### PR TITLE
Added methods to get actual tweet likes and retweets

### PR DESCRIPTION
**Pull request by bengabp**
* ### Get tweet actual retweets
    - `Twitter().tweet_retweeters(identifier)`  returns a Generator of Dictionaries which are the actual retweets

* ### Get tweet actual likes
    - `Twitter().tweet_likes(identifier)` returns a Generator of Dictionaties which are the actual likes

* ### New exception class 
     - `RateLimitError(Exception)` a class for the rate limit exception from the twitter frontend api (`error_code 429`) , this is rarely seen and can be triggered by making 20k-25K requests or more in a short time. Users have to wait for 15minutes before they can make requests again.

### CODE REVIEW

**Comment by bengabp** [comment]
Ive also added a demo script `demo.py` to test my changes. Please keep in mind that you need authenticated cookies for this to work.

File: src/tweety/bot.py [hunk 1]
```diff
@@ -95,20 +98,21 @@ def get_tweets(self, username: Union[str, int, User], pages: int = 1, replies: b
...
-        user_id = self._get_user_id(username)
+		user_id = self._get_user_id(username)
 
-        userTweets = UserTweets(user_id, self.request, pages, replies, wait_time, cursor)
+		userTweets = UserTweets(user_id, self.request, pages, replies, wait_time, cursor)
 
-        # TODO : Find proper way to run the generator
-        results = [i for i in userTweets.generator()]
+		# TODO : Find proper way to run the generator
+		results = [i for i in userTweets.generator()]
```
Comment (philipperemy) [inline src/tweety/bot.py:109]: you can probably do `list(userTweets.generator())` instead.

File: src/tweety/bot.py [hunk 2]
```diff
@@ -119,41 +123,43 @@ def iter_tweets(self,  username: Union[str, int, User], pages: int = 1, replies:
...
+			data = {
+				"name": i['item']['content']['trend']['name'],
+				"url": str(i['item']['content']['trend']['url']['url']).replace("twitter://",
+																				"https://twitter.com/").replace("query",
+																												"q"),
+			}
+			try:
+				if i['item']['content']['trend']['trendMetadata']['metaDescription']:
+					data['tweet_count'] = i['item']['content']['trend']['trendMetadata']['metaDescription']
+			except:
```
Comment (philipperemy) [inline src/tweety/bot.py:155]: If you know which Exception is thrown better to add it here.

Comment (bengabp) [reply src/tweety/bot.py:155]: I am not sure i wrote that code block, you did

**Comment by philipperemy** [comment]
For info the state of the PR does not really solve the issue:

```
INFO:httpx:HTTP Request: POST https://api.twitter.com/1.1/guest/activate.json "HTTP/1.1 200 OK"
INFO:httpx:HTTP Request: GET https://twitter.com/i/api/graphql/WzJjibAcDa-oCjCcLOotcg/UserTweets?variables=%7B%22userId%22%3A+%2244196397%22%2C+%22count%22%3A+40%2C+%22includePromotedContent%22%3A+true%2C+%22withQuickPromoteEligibilityTweetFields%22%3A+true%2C+%22withVoice%22%3A+true%2C+%22withV2Timeline%22%3A+true%7D&features=%7B%22rweb_lists_timeline_redesign_enabled%22%3A+false%2C+%22blue_business_profile_image_shape_enabled%22%3A+true%2C+%22responsive_web_graphql_exclude_directive_enabled%22%3A+true%2C+%22verified_phone_label_enabled%22%3A+false%2C+%22creator_subscriptions_tweet_preview_api_enabled%22%3A+false%2C+%22responsive_web_graphql_timeline_navigation_enabled%22%3A+true%2C+%22responsive_web_graphql_skip_user_profile_image_extensions_enabled%22%3A+false%2C+%22tweetypie_unmention_optimization_enabled%22%3A+true%2C+%22vibe_api_enabled%22%3A+true%2C+%22responsive_web_edit_tweet_api_enabled%22%3A+true%2C+%22graphql_is_translatable_rweb_tweet_is_translatable_enabled%22%3A+true%2C+%22view_counts_everywhere_api_enabled%22%3A+true%2C+%22longform_notetweets_consumption_enabled%22%3A+true%2C+%22tweet_awards_web_tipping_enabled%22%3A+false%2C+%22freedom_of_speech_not_reach_fetch_enabled%22%3A+true%2C+%22standardized_nudges_misinfo%22%3A+true%2C+%22tweet_with_visibility_results_prefer_gql_limited_actions_policy_enabled%22%3A+false%2C+%22interactive_text_enabled%22%3A+true%2C+%22responsive_web_text_conversations_enabled%22%3A+false%2C+%22longform_notetweets_rich_text_read_enabled%22%3A+true%2C+%22longform_notetweets_inline_media_enabled%22%3A+false%2C+%22responsive_web_enhance_cards_enabled%22%3A+false%7D "HTTP/1.1 404 Not Found"
Traceback (most recent call last):
  File "/private/tmp/trades/ff.py", line 31, in <module>
    main()
  File "/private/tmp/trades/ff.py", line 22, in main
    raise e
  File "/private/tmp/trades/ff.py", line 17, in main
    tweet = list(app.get_tweets(user_id, pages=1))[0]
  File "/private/tmp/tweety/src/tweety/bot.py", line 109, in get_tweets
    results = [i for i in userTweets.generator()]
  File "/private/tmp/tweety/src/tweety/bot.py", line 109, in <listcomp>
    results = [i for i in userTweets.generator()]
  File "/private/tmp/tweety/src/tweety/types/usertweet.py", line 75, in generator
    _, tweets = self.get_next_page()
  File "/private/tmp/tweety/src/tweety/types/usertweet.py", line 42, in get_next_page
    response = self.http.get_tweets(self.user_id, replies=self.get_replies, cursor=self.cursor)
  File "/private/tmp/tweety/src/tweety/http.py", line 121, in get_tweets
    response = self.__get_response__(**request_data)
  File "/private/tmp/tweety/src/tweety/http.py", line 35, in __get_response__
    raise UnknownError(
tweety.exceptions_.UnknownError: Unknown Error Occurs on Twitter
```

```python
import datetime
import logging
import time

from tweety.bot import Twitter

def main():
    logging.basicConfig(level=logging.INFO)
    app = Twitter()
    user_id = 44196397
    tweet = list(app.get_tweets(user_id, pages=1))[0]

if __name__ == '__main__':
    main()
```

**Comment by bengabp** [comment]
@philipperemy I know, I actually have solutions to these issues but I lack collaborative skills, and I dont know who is working on any feature to avoid me from building what has already been built by other devs working on this. Also the maintainer has not given any reply based on when he will be releasing a new version or upgrading the library. I think he should consider creating a discord server or google chat space, where we can share updates more frequently as supposed to creating multiple related issues.

**Comment by mahrtayyab** [comment]
It is fixed , please wait for some hours so I can complete my testing

**Comment by philipperemy** [comment]
@mahrtayyab thank you very much. I'll stand by. Take your time.
@bengabp Understood :)

File: src/tweety/bot.py (hunk 2 above)
Comment (bengabp) [reply src/tweety/bot.py:155]: @philipperemy i think this was written by the maintainer, but its likely a KeyError or IndexError that would occur, i encountered it while adding the features

**Comment by mahrtayyab** [comment]
@bengabp would it not be better to add these methods to the actual Tweet class?

**Comment by mahrtayyab** [comment]
Because we need to check for the tweet existence else raise the Exception

**Comment by bengabp** [comment]
> @bengabp would it not be better to add these methods to the actual Tweet class?

Can you do it, I may not be having enough time to work on this project as I have a hand-full of tasks to complete in my team

**Pull request closed without merging**